zerver/lib/markdown/__init__.py

```diff
@@ -897,9 +899,9 @@ def twitter_text(
                 for match in re.finditer(re.escape(short_url), text, re.IGNORECASE)
             )
         # Build dicts for emojis
-        for match in re.finditer(UNICODE_EMOJI_RE, text, re.IGNORECASE):
+        for match in regex.finditer(POSSIBLE_EMOJI_RE, text, regex.IGNORECASE):
```

Comment:
This use of `POSSIBLE_EMOJI_RE` is not expecting the `(.*?)` we prepended and appended. I think we need to define `POSSIBLE_EMOJI_RE` without them and `POSSIBLE_EMOJI_WHOLE_STRING_RE = regex.compile(rf"^(.*?){POSSIBLE_EMOJI_RE.pattern}(.*?)$", regex.VERBOSE | regex.DOTALL)` to pass to the `UnicodeEmoji` constructor.

Also, this can be simplified to `POSSIBLE_EMOJI_RE.finditer(text)`.

Reply:
I'm having trouble getting that working -- maybe we do need that `(.*?)` over the whole twitter text?

I added the simplification -- did you drop `re.IGNORECASE` on purpose?

Reply:
Before your PR, `UNICODE_EMOJI_RE` does not have the `(.*?)`s, so your PR is incorrectly adding it here. It does not make sense to use `finditer` on a regex that matches the whole string.

`re.IGNORECASE` is not important here; there’s nothing uppercase or lowercase about Unicode emoji. It was probably cargo-culted from the other calls (where it also doesn’t seem important, but that’s a question for another time).